### PR TITLE
[Impeller] Wire-up AndroidSurfaceImpellerVulkan

### PR DESCRIPTION
**Pull request by iskakaushik**
- This enables preliminary support for Impeller + Vulkan on Android.
- Pending multiple fidelity fixes and adding a preference for Vulkan.
- Also noteworthy is https://github.com/flutter/flutter/issues/114531

### CODE REVIEW

File: shell/platform/android/platform_view_android.cc [hunk 1]
```diff
@@ -43,8 +44,15 @@ std::unique_ptr<AndroidSurface> AndroidSurfaceFactoryImpl::CreateSurface() {
                                                       jni_facade_);
     case AndroidRenderingAPI::kOpenGLES:
       if (enable_impeller_) {
+// TODO(kaushikiska@): Enable this after wiring a preference for Vulkan backend.
+#if false
```
Comment (chinmaygarde) [inline shell/platform/android/platform_view_android.cc:48]: Hmm, `false` is a preprocessor define? TIL

File: tools/gn [hunk 2]
```diff
@@ -436,10 +436,10 @@ def to_gn_args(args):
     gn_args['skia_use_metal'] = True
     gn_args['shell_enable_metal'] = True
 
-  # Enable Vulkan on all platforms except for Android and iOS. This is just
+  # Enable Vulkan on all platforms except for iOS. This is just
```
Comment (chinmaygarde) [inline tools/gn:439]: 🥳

**Review by chinmaygarde — APPROVED** [review]

**Comment by dnfield** [comment]
before: https://storage.googleapis.com/flutter_infra_release/flutter/edbba91086317574ae116482841633912245fb86/android-arm64-release/sizes/index.html

after: https://storage.googleapis.com/flutter_infra_release/flutter/c7b2230e98d28c8538faeabed42c27145ae30a68/android-arm64-release/sizes/index.html

Seems like a bunch of things went up by ~1kb, but the treemap doesn't seem to reflec tthe size increase you were mentioning...

**Comment by chinmaygarde** [comment]
> Seems like a bunch of things went up by ~1kb, but the treemap doesn't seem to reflec tthe size increase you were mentioning...

Perhaps the compression is less effective?

**Comment by chinmaygarde** [comment]
Umm, is that the right treemap? I don't see `impeller_entity_shaders_vk_data` (or something similar) in the large symbols list in the after trace. I do see `impeller_entity_shaders_gles_data` though.

**Comment by iskakaushik** [comment]
@chinmaygarde could it be the case that this could have gotten tree-shaken away because it's not actually wired yet? The 
 entry point is behind `#if false`. https://github.com/flutter/engine/pull/37249/files#diff-033d9ee4b0ca06755bdc597c70b48f7e46d73a5234f9964a982cb383046c0ab4R48

**Comment by chinmaygarde** [comment]
Oh, right. Yeah. That makes sense. We should see the size increases after we actually start using it.

**Comment by dnfield** [comment]
Probably it.

**Comment by zanderso** [comment]
SkiaPerf is also reporting no size increase https://flutter-flutter-perf.skia.org/e/?queries=sub_result%3Dlibflutter_compressed_bytes%26test%3Dhello_world_android__compile

**Pull request merged**